en/tools/calc-size.md

```diff
@@ -0,0 +1,184 @@
+      + 1      // redeemScript length indicator
+      + (1 + compressed_pubkey) * 3 // pubkeys (including pushes)
+      + 3 // OP_2 OP_3 OP_CHECKMULTISIG
+      + nsequence
+    ),
+    "output": (
+        nvalue
+      + 1 // scriptPubKey length
+      + 2 // OP_CHECKHASHVERIFY OP_CHECKEQUAL
+      + 1 + ripemd160 // includes push
+    ),
+    "segwit": false
+  };
+
+  types["p2wsh2_3"] = {
+    "input": (
+      outpoint
+      +  1 // scriptSig length
+      + (  // Witness elements
+          1       // OP_0
+        + (1 + ecdsa_signature) * 2  // signatures (including length indicators)
+        + 1      // witnessScript length indicator
+        + (1 + compressed_pubkey) * 3 // pubkeys (including pushes)
+        + 3 // OP_2 OP_3 OP_CHECKMULTISIG
+      )/segwit_discount
+      + nsequence
+    ),
+    "output": (
+        nvalue // nValue
+      + 1 // scriptPubKey length
+      + 1 // OP_0
+      + 1 + sha256 // includes push
+    ),
+    "segwit": true
+  };
+
+  types["p2tr"] = {
+    "input": (
+      outpoint
+      +  1 // scriptSig length
+      + (  // Witness elements
+          1 + schnorr_signature // includes length indicator
+      )/segwit_discount
+      + nsequence
+    ),
+    "output": (
+         nvalue
+      +  1 // scriptPubKey length
+      +  1 // OP_1
+      + 1 + taproot_pubkey // includes push
+    ),
+    "segwit": true
+  };
+
+  // Get the form
+  var form = document.forms["vbytescalc"];
+
+  // The input/output type
+  // TODO: maybe allow the input type to be different from the output type
+  var type = form.elements["type"];
+  input_size = types[type.value].input;
+  output_size = types[type.value].output;
+
+  // The number of inputs and outptus
+  var inputs = form.elements["inputs"].value;
+  if (inputs != "") {
+    inputs = parseInt(inputs);
+  } else {
+    inputs = 0;
+  }
+  var outputs = form.elements["outputs"].value;
+  if (outputs != "") {
+    outputs = parseInt(outputs);
+  } else {
+    outputs = 0;
+  }
+
+  // Calculate the transaction's overhead (the size indpendent of inputs and outputs)
+  if (types[type.value].segwit == true) {
+    witness_flag = 0.75;  // segwit marker, segwit flag, segwit # of witness elemnets
+  } else  {
+    witness_flag = 0;
+  }
+  overhead = (
+      4 // nVersion
+    + 1 // number of inputs, TODO: increase for >252 inputs
+    + 1 // number of outputs, TODO: increase for >252 outputs
+    + 4 // nLockTime
+    + witness_flag
+  )
+
+  size = overhead + input_size * inputs + output_size * outputs;
+  document.getElementById('result').innerHTML = (
+    "Total size: " + size + "<br>"
+    + "Overhead: " + overhead + "<br>"
+    + "Input size = " + inputs + " x " + input_size + ' = ' + inputs * input_size + "<br>"
+    + "Output size: " + outputs + " x " + output_size + " = " + outputs * output_size
+  );
+}
+
+// Calculate the result for the default form parameters
+calculateTotal();
```

Comment:
Id suggest a default input output count here (say 1 in 2 out?) to make initial page load more informative to a new visitor

Reply:
Perfecto